File: languages/java/src/main/java/com/bloock/sdk/entity/identity_v2/DatetimeAttributeDescriptor.java
package com.bloock.sdk.entity.identity_v2;

import com.bloock.sdk.bridge.proto.IdentityEntitiesV2;

/**
 * Represents a descriptor for an attribute with a datetime value.
 */
public class DatetimeAttributeDescriptor extends AttributeDescriptor {
  /**
   * Constructs an DatetimeAttributeDescriptor object with the specified parameters.
   * @param displayName
   * @param technicalName
   * @param description
   * @param required
   */
  public DatetimeAttributeDescriptor(
      String displayName, String technicalName, String description, Boolean required) {
    super(displayName, technicalName, description, required);
  }

  public static DatetimeAttributeDescriptor fromProto(
      IdentityEntitiesV2.DateTimeAttributeDefinitionV2 res) {
    return new DatetimeAttributeDescriptor(
        res.getDisplayName(), res.getId(), res.getDescription(), res.getRequired());
  }

  public IdentityEntitiesV2.DateTimeAttributeDefinitionV2 toProto() {
    return IdentityEntitiesV2.DateTimeAttributeDefinitionV2.newBuilder()
        .setDisplayName(this.displayName)
        .setId(this.technicalName)
        .setDescription(this.description)
        .setRequired(this.required)
        .build();
  }
}

File: languages/php/Bloock/Entity/IdentityV2/DatetimeAttributeDescriptor.php
<?php

namespace Bloock\Entity\IdentityV2;

use Bloock\DateTimeAttributeDefinitionV2;

class DatetimeAttributeDescriptor extends AttributeDescriptor
{
    public static function fromProto(DateTimeAttributeDefinitionV2 $res): DatetimeAttributeDescriptor
    {
        return new DatetimeAttributeDescriptor($res->getDisplayName(), $res->getId(), $res->getDescription(), $res->getRequired());
    }

    public function toProto(): DateTimeAttributeDefinitionV2
    {
        $p = new DateTimeAttributeDefinitionV2();
        $p->setDisplayName($this->displayName);
        $p->setId($this->technicalName);
        $p->setDescription($this->description);
        $p->setRequired($this->required);
        return $p;
    }
}